scripts/xtensa-build-zephyr.py

```diff
@@ -902,6 +908,80 @@ def build_platforms():
 			symlinks=True, ignore_dangling_symlinks=True, dirs_exist_ok=True)
 
 
+def install_lib(sof_lib_dir, abs_build_dir):
+	"""[summary] Sign loadable llext modules, if any, copy them to the
+	deployment tree and create UUID links for the kernel to find and load
+	them."""
+
+	global signing_key
+	global platform_wconfig
+
+	with os.scandir(str(abs_build_dir)) as iter:
+		rimage_executable = platform_wconfig.get("rimage.path")
+		if rimage_executable is None:
+			rimage_executable = platform_wconfig.get("rimage.path", None,
+								 west_config.ConfigFile.LOCAL)
+
+		if args.key_type_subdir != "none":
+			sof_lib_dir = sof_lib_dir / args.key_type_subdir
+
+		sof_lib_dir.mkdir(parents=True, exist_ok=True)
+
+		for entry in iter:
+			if (not entry.is_dir or
+			    not entry.name.endswith('_llext')):
+				continue
+
+			entry_path = pathlib.Path(entry.path)
+
+			uuids = entry_path / 'llext.uuid'
+			if not os.path.exists(uuids):
+				print(f"Directory {entry.name} has no llext.uuid file. Skipping.")
+				continue
+
+			# replace '_llext' with '.llext', e.g.
+			# eq_iir_llext/eq_iir.llext
+			llext_base = entry.name[:-6]
+			llext_file = llext_base + '.llext'
+
+			dst = sof_lib_dir / llext_file
+
+			rimage_cfg = entry_path / 'rimage_config.toml'
+			llext_input = entry_path / (llext_base + '.so')
+			llext_output = entry_path / llext_file
+
+			sign_cmd = [str(rimage_executable), "-o", str(llext_output),
+				    "-e", "-c", str(rimage_cfg),
+				    "-k", str(signing_key), "-l", "-r",
+				    str(llext_input)]
+			execute_command(sign_cmd, cwd=west_top)
+
+			# Explicitly rely on default exception handling
+			fdst = open(dst, 'wb')
+			fllext = open(llext_output, 'rb')
+			fman = open(llext_output.with_suffix('.llext.xman'), 'rb')
+
+			# Concatenate the manifest and the llext
+			shutil.copyfileobj(fman, fdst)
+			shutil.copyfileobj(fllext, fdst)
+
+			fdst.close()
+			fman.close()
+			fllext.close()
+
+			# Create symbolic links for all UUIDs
+			with open(uuids, 'r') as f_llext:
+				lines = f_llext.readlines()
+				for line in lines:
+					linkname = line.strip() + '.bin'
+					link = sof_lib_dir / linkname
+					if (os.path.exists(link)):
+						link.unlink()
+					symlink_or_copy(sof_lib_dir, llext_file,
+							sof_lib_dir, linkname)
+				f_llext.close()
```

Comment:
@marc-hb ah, there you go - I didn't know leaving `with` was releasing resources, thanks!

Reply:
huh, windows Python version seems to have a problem with that... https://github.com/thesofproject/sof/actions/runs/8936988225/job/24548392500?pr=9013

Reply:
@marc-hb I removed parentheses for which I had to put the whole `with` statement on 1 line and now windows are happy. I.e.
```
with open(x1) as f1, open(x2) as f2, open(x3) as f3:
```
and now the line is 150 characters long.

Reply:
I use the "black" Python formatter to solve all problems like this one:
```
apt install black
cp scripts/xtensa-build-zephyr.py for_black.py
black for_black.py # reformats everything in place
```

... then open `for_black.py` and look at how `black` solved the problem.

Reply:
> huh, windows Python version seems to have a problem with that... https://github.com/thesofproject/sof/actions/runs/8936988225/job/24548392500?pr=9013

Wow, I've never seen any Linux/Windows difference like this one, amazing. Maybe some CRLF or weird tab/space mix?!

Anyway it's gone now!

Reply:
> > huh, windows Python version seems to have a problem with that... https://github.com/thesofproject/sof/actions/runs/8936988225/job/24548392500?pr=9013
> 
> Wow, I've never seen any Linux/Windows difference like this one, amazing. Maybe some CRLF or weird tab/space mix?!
> 
> Anyway it's gone now!

@marc-hb it's gone now, because I put it all in one line, right?